### DISCUSSION TITLE
[x] Large Language Models - How's it going?

### DISCUSSION THREAD

**Discussion by dewoller** [RSEAA/RSEAA.github.io#53] [Discussion topics for #RSEAA2023]
I have been working pretty intensively with Large Language Models (LLM) for the last nine months.  Our work as software engineers has the potential to be fundamentally changed by these tools.  I really appreciate the way LLM's have helped my software development practice, among  many other things.  It has been a long time since I was astonished *every* day!

I propose to lead and moderate a structured discussion, where everyone gets a chance to be listened to, around the following three topics:
-  LLM  success stories - where has it gone well?
-  LLM difficulties - What is a bit hard still?
-  Your  future with LLM: where is it going?

Myself, I have mostly worked with state government covid data for the past 3 years, and spent the last 3 months getting chatGPT to mark papers. 

Please comment below if you have any questions or suggestions.

Dennis Wollersheim (PhD)

**Comment by manodeep** [comment]
I tried out ChatGPT for generating complex SIMD code, and there were a lot of subtle errors. But for a more straightforward task (buffered IO) ChatGPT produced close-to-production code. I wonder if others have had similar experiences.

**Comment by abickerstaffe** [comment]
Chatbots - concerns over advice given and potential legal ramifications (organisational liability)

**Comment by rowlandm** [comment]
We are looking at a Data Commons to put in multi-omics data plus spatial omics and then try to use LLMs to query data.

**Reply by rowlandm** [reply]
Looking at Hugging face etc.

**Comment by abickerstaffe** [comment]
Importance of training data quality.
Domain-specific training data - who prepares it, who QCs it, how much work is involved?

**Reply by rowlandm** [reply]
Even data standardisation is a thing.

**Comment by abickerstaffe** [comment]
Academic aspects - writing and reviewing grants, papers, presentations, etc. What is OK, what is not OK?

**Comment by manodeep** [comment]
Astronomy LLM paper came out today: https://arxiv.org/abs/2309.06126

https://huggingface.co/universeTBD/astrollama

**Comment by rowlandm** [comment]
Using LLMs for codign is like asking novice programmers for help.

**Comment by mcmero** [comment]
I've found ChatGPT great for synthesising information, particularly where Google searches can produce very scattershot results requiring lots of digging for some key bit of info. One barrier I've run up against quite often is the cutoff date of the trained model (Sep 2021). Being in the technology area, it would be really useful to get more up-to-date models regularly.

**Comment by rowlandm** [comment]
Useful for asking for "key data" in a scientific field - eg. what is the best data to test cryo-em on? It mentioned apoferritin

**Comment by rowlandm** [comment]
Co-pilot a plugin for LLM development environtment eg. VS code.

**Reply by manodeep** [reply]
I have used Copilot and found it quite helpful (and almost never wrong)

**Comment by abickerstaffe** [comment]
Can be used for teaching, helpful for learning different programming languages.

**Comment by kiowa-scott-hurley** [comment]
The analogy of LLMs as a JPEG is a really helpful way of thinking about how "useful" LLMs are for a given task: 
https://www.newyorker.com/tech/annals-of-technology/chatgpt-is-a-blurry-jpeg-of-the-web

I've been using LLMS in anger across a breadth of domains and this "idea" holds true across use-cases.

**Reply by manodeep** [reply]
Hahahahahaha "using LLMs in anger"
